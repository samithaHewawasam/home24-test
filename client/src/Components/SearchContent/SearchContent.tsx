import { Empty, Layout, Row, Skeleton } from 'antd'
import { ArticleCard } from 'Components/ArticleCard'
import { querySearch } from 'GraphRepository'
import { useGQLRequest } from 'Http'
import { Article } from 'Interfaces'
import React, { memo, useCallback, useEffect } from 'react'
import { useSearchParams } from 'react-router-dom'

const { Content } = Layout

const SearchContent: React.FC = memo(() => {
  const [searchParams] = useSearchParams()
  const {
    data,
    isError = false,
    isLoading = false,
    setQuery
  } = useGQLRequest<{
    categories: any
  }>({
    data: { categories: { filters: [] } },
    isError: false,
    isLoading: false
  })

  if (data === null && isError) {
    return <Empty style={{ margin: '20vh auto' }} />
  }
  const {
    categories: { filters }
  } = data

  useEffect(() => {
    if (searchParams.get('query') === null) return

    if (typeof setQuery !== 'undefined') {
      setQuery({
        query: querySearch,
        variables: {
          locale: 'de_DE',
          query: searchParams.get('query'),
          urlParams: `query=${searchParams.get('query') ?? ''}`
        }
      })
    }
  }, [searchParams])

  const handleCartUpdate: (article: Article) => void = useCallback(
    /**
     * @param {Article} article - Article object.
     * @description set articles to cart item
     */

    (article: Article): void => {
      // setCartItems((prevState) => {
      //   return [...prevState, article]
      // })
    },
    [filters]
  )

  return (
    <Content style={{ margin: '24px 16px 0' }}>
      <Skeleton loading={isLoading}>
        <Row gutter={[16, 16]}>
          {filters.length > 0 &&
            filters.map((article: any, index: number) => {
              return (
                <ArticleCard
                  article={{
                    ...article,
                    index,
                    images: [{ url: '' }],
                    prices: { regular: { value: 0 } }
                  }}
                  key={index}
                  handleCartUpdate={handleCartUpdate}
                />
              )
            })}
        </Row>
      </Skeleton>
    </Content>
  )
})

SearchContent.displayName = 'SearchContent'
export default SearchContent
