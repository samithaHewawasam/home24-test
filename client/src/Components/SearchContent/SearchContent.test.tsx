import { cleanup, render } from '@testing-library/react'
import { act } from '@testing-library/react-hooks/dom'
import React from 'react'
import { BrowserRouter } from 'react-router-dom'
import renderer from 'react-test-renderer'
import SearchContent from './SearchContent'

/**
 * @description set window object with search query param
 */
const location = {
  ...window.location,
  search: '?query=yonkers'
}
Object.defineProperty(window, 'location', {
  writable: true,
  value: location
})

describe('Search Content', () => {
  const SearchContentComponent = (
    <BrowserRouter>
      <SearchContent />
    </BrowserRouter>
  )
  beforeEach(async () => {
    await act(async () => {
      render(SearchContentComponent)
    })
  })

  afterEach(() => {
    cleanup()
  })

  /**
   * @description SnapShot for Search Content
   */
  it('renders Search Content correctly', () => {
    const tree = renderer.create(SearchContentComponent).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('render the AutoComplete', () => {
    // TODO - Write test for Search Suggestions
    // TODO - Write test for Click and navigate to the search route
  })
})
