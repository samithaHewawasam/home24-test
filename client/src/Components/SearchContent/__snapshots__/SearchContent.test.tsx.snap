// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Content renders Search Content correctly 1`] = `
<main
  className="ant-layout-content"
  style={
    Object {
      "margin": "24px 16px 0",
    }
  }
>
  <div
    className="ant-row"
    style={
      Object {
        "marginBottom": -8,
        "marginLeft": -8,
        "marginRight": -8,
        "marginTop": -8,
      }
    }
  />
</main>
`;
